packages/dashboard/src/redux/hooks.ts

```diff
@@ -0,0 +1,19 @@
+import { useDispatch, useSelector } from 'react-redux';
+import { PluginData, RootState } from '@deephaven/redux';
+import { getPluginDataForDashboard } from './selectors';
+import { setDashboardPluginData } from './actions';
+
+export function useDashboardPluginData(
+  dashboardId: string,
+  pluginId: string
+): [PluginData, (data: PluginData) => void] {
+  const dispatch = useDispatch();
+  return [
+    useSelector((store: RootState) =>
+      getPluginDataForDashboard(store, dashboardId, pluginId)
+    ),
+    data => dispatch(setDashboardPluginData(dashboardId, pluginId, data)),
+  ];
```

Comment:
I typically would explicitly put the `useSelector` at the top level. Also we should use `useCallback` for the data callback, and we need JS docs on the parameters passed in and what the return values are. e.g.
```
const dispatch = useDispatch();
const data = useSelector( ... );
const setData = useCallback(newData => dispatch(...), [dashboardId, pluginId, dispatch]);
return [ data, setData ];
```